### ISSUE TITLE
Update documentation to make it clearer CocoaPods is not mandatory

### ISSUE THREAD

**Issue by tfcporciuncula** [getsentry/sentry-kotlin-multiplatform#298]
### Problem Statement

Currently, the docs on https://docs.sentry.io/platforms/kotlin-multiplatform suggest CocoaPods is mandatory when integrating the library. There's even a section on SPM at the bottom that says we can use SPM for other things, but we still need CocoaPods for Sentry.

Is this really true, though? We tried adding Sentry directly on iOS with SPM (https://docs.sentry.io/platforms/apple/install/swift-package-manager) and it seems to have worked fine. Looking at https://github.com/getsentry/sentry-kotlin-multiplatform/issues/118 and https://github.com/getsentry/sentry-kotlin-multiplatform/issues/108 it also seems to confirm this strategy would be fine.

It would be great to get clarity on this and get the docs updated if they're indeed misleading.

**Comment by buenaflor** [comment]
I assume your `is Static = true` is set in the framework? in that case it should work but not for tests because the linker is not configured there, that's the main issue. When you look at the [cocoapods plugin source code](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/native/cocoapods/KotlinCocoapodsPlugin.kt#L604C1-L615C1) they set up the linking automatically that's why it's suggested to set up both. If you don't setup linking [you will get this error](https://github.com/getsentry/sentry-kotlin-multiplatform/issues/176)

Regardless we actually have new docs merged in today (not sure when they will be live) that suggests to use our gradle plugin that does all the linking so you're not tied to using only cocoapods. Please try it out

`plugins { id("io.sentry.kotlin.multiplatform.gradle") version "0.10.0" }`

**Comment by tfcporciuncula** [comment]
Yes, we do set `isStatic = true`! And thanks, we'll try out the plugin and report back.

**Comment by buenaflor** [comment]
Closing as new docs now states using the gralde plugin instead which handles the configuration automatically
